controllers/operatorpolicy_controller.go

```diff
@@ -733,6 +738,65 @@ func (r *OperatorPolicyReconciler) usingExistingSubIfFound(
 	return false
 }
 
+// resolveVersionsTemplates will resolve all templates in spec.versions. Any versions that resolve to an empty string
+// are discarded.
+func resolveVersionsTemplates(
+	policy *policyv1beta1.OperatorPolicy, tmplResolver *templates.TemplateResolver,
+) error {
+	if tmplResolver == nil {
+		return nil
+	}
+
+	var hasTemplate bool
+
+	for _, version := range policy.Spec.Versions {
+		if templates.HasTemplate([]byte(version), "", false) {
+			hasTemplate = true
+
+			break
+		}
+	}
+
+	if !hasTemplate {
+		return nil
```

Comment:
Should we remove the requirement that items in the `versions` list must be non-empty? Since we'll allow them to be empty after template resolution, it might be nice to consistently allow them without templates.

The question is whether these should all be allowed and behave the same:
- `[]`
- `[""]`
- `["{{ /* comment */ }}"]`
- `["{{ if false }}fake{{ end }}"]`

Reply:
That's a good point. I'll adjust it.